src/client/localplayer.cpp

```diff
@@ -442,13 +442,19 @@ void LocalPlayer::move(f32 dtime, Environment *env, f32 pos_max_d,
 			// Move player to the maximal height when falling or when
 			// the ledge is climbed on the next step.
 
-			// Smoothen the movement (based on 'position.Y = bmax.Y')
-			// Check if there is space for sneak glitch
 			v3f pos(position.X, position.Y + y_diff * dtime * 22.0f + BS * 0.01f, position.Z);
-			if (!collision_check_intersection(env, m_client, m_collisionbox, pos)) {
+			if (y_diff < BS * 0.6f) {
+				// Smoothen the movement (based on 'position.Y = bmax.Y')
 				position.Y = std::min(pos.Y, bmax.Y);
 				m_speed.Y = 0.0f;
 			}
+			else {
+				// Check if there is space for sneak glitch
+				if (!collision_check_intersection(env, m_client, m_collisionbox, pos)) {
+					position.Y = std::min(pos.Y, bmax.Y);
+					m_speed.Y = 0.0f;
```

Comment:
These two lines are duplicated too, it should be possible to have them in one `if`